### PR TITLE
collect_items and Vroom can handle shipment relations

### PR DESCRIPTION
**Pull request by senhalil**
- collect_data_items respects the minimum vehicle limits
- linking_relations can be grouped in collect_items
- split_solve represents services w multiple points
- Vroom can handle simplified setup durations
- Vroom can handle shipments as relation
- max_split_size prevents only optim-sync and Vroom can be called for sub-rpoblems
- small quality and linter fixes

### CODE REVIEW

File: wrappers/wrapper.rb [hunk 1]
```diff
@@ -1111,6 +1113,139 @@ def simplify_vehicle_pause(vrp, result = nil, options = { mode: :simplify })
...
+          vehicle[:simplified_coef_setup] = nil
+          vehicle[:simplified_additional_setup] = nil
+        }
+      when :patch_result
+        # patches the result
+        # the travel_times need to be decreased and setup_duration need to be increased by
+        # (coef_setup * setup_duration + additional_setup) if setup_duration > 0 and travel_time > 0
+        return nil unless vrp.services.any?{ |s| s[:simplified_setup_duration] }
+
+        vehicles_grouped_by_vehicle_id = vrp.vehicles.group_by(&:id)
```
Comment (braktar) [inline wrappers/wrapper.rb:1209]: can we group by id ? I assume it is supposed to be unique. original_id may not be unique

I you are expecting to have exactly one vehicle by group, I would be expecting a hash id=>vehicle.

**Review by braktar — APPROVED** [review]

File: lib/interpreters/split_clustering.rb [hunk 2]
```diff
@@ -1162,26 +1279,27 @@ def remove_used_empties_and_refills(vrp, result)
...
-        returned = if node.nil?
-          []
         elsif (graph[node][:level]).zero?
           [node]
         else
           [tree_leafs(graph, graph[node][:left]), tree_leafs(graph, graph[node][:right])]
         end
+      end
+
+      def tree_leafs_delete(graph, node)
```
Comment (fonsecadeline) [inline lib/interpreters/split_clustering.rb:1289]: _tree_leafs_delete_ and _tree_leafs_ seem very similar. Could we consider having only _tree_leafs_ function but with an option to delete ?

Comment (senhalil) [reply lib/interpreters/split_clustering.rb:1289]: I haven't modified `tree_leafs_delete` in purpose, it shouldn't be in my dev at all :confused: I am confused!

Comment (senhalil) [reply lib/interpreters/split_clustering.rb:1289]: Oh it is the syntax correction. I apparently corrected the `if` conditional assignment but the dev has nothing to do with `tree_leafs` functions. They are used in `split_hierarchical`. So I would rather not make such a change in this PR.

Comment (fonsecadeline) [reply lib/interpreters/split_clustering.rb:1289]: In fact I could not see the difference :p maybe some spaces where removed ?

Comment (senhalil) [reply lib/interpreters/split_clustering.rb:1289]: If you check with "Hide whitespace changes" option of the review page, then the diff becomes more obvious and minimal.

**Review by fonsecadeline — APPROVED** [review]

**Pull request merged**